src/datomish/datom.cljc

```diff
@@ -0,0 +1,137 @@
+;; This Source Code Form is subject to the terms of the Mozilla Public
+;; License, v. 2.0. If a copy of the MPL was not distributed with this
+;; file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+;; Purloined from DataScript.
```

Comment:
@ncalexan: are we allowed to use Eclipse Public License code? Does it affect whether we are MPL or Apache licensed?